core/src/com/unciv/logic/GameStarter.kt

```diff
@@ -197,9 +197,14 @@ object GameStarter {
         val allMercantileResources = ruleset.tileResources.values.filter { it.unique == "Can only be created by Mercantile City-States" }.map { it.name }
         val unusedMercantileResources = Stack<String>()
         unusedMercantileResources.addAll(allMercantileResources.shuffled())
+        var numbersToTake = newGameParameters.numberOfCityStates
 
-        for (cityStateName in availableCityStatesNames.take(newGameParameters.numberOfCityStates)) {
+        for (cityStateName in availableCityStatesNames.take(numbersToTake)) {
             val civ = CivilizationInfo(cityStateName)
+            if (civ.cityStateType == CityStateType.Religious && !newGameParameters.religionEnabled){
```

Comment:
I'd still advise on using `gameInfo.hasReligionEnabled()` instead of `!newGameParameters.religionEnabled` as it prevents code duplication, and also provides correct behaviour when starting in the industrial era or later.

Reply:
if i use the gameInfo class, I will get a horde of uninitialized variable errors

Reply:
I thought you could work around them, but looking closer that does seem impossible. I'll check out later today why the transients of the gameInfo aren't already set by this point, as especially the ruleSet should already be known at this point.

Reply:
Alright, it's fine to add the line:
```gameInfo.ruleSet = ruleset // need to set this transient so that we can get the starting era```
Just before calling `addCivilizations` in `startNewGame` (line 60 or so), and that should fix all the `unitialized variable` errors

Reply:
fixed

Reply:
@xlenstra some other error is coming in the unit test